be/extmap.c

```diff
@@ -131,14 +132,19 @@ static struct m0_rwlock *emap_rwlock(struct m0_be_emap *emap)
 
 static void emap_dump(struct m0_be_emap_cursor *it)
 {
-	struct m0_be_emap_cursor scan;
-	struct m0_uint128       *prefix = &it->ec_key.ek_prefix;
-	struct m0_be_emap_seg   *seg    = &scan.ec_seg;
-	int                      i;
-	int                      rc;
+	int                       i;
+	int                       rc;
+	struct m0_be_emap_cursor *scan;
+	struct m0_uint128        *prefix = &it->ec_key.ek_prefix;
+	struct m0_be_emap_seg    *seg;
+
+	M0_ALLOC_PTR(scan);
```

Comment:
is really necessary to allocate locally here ?

Reply:
Yes, to decrease the stack size.